src/Orleans/Configuration/ProviderConfiguration.cs

```diff
@@ -193,23 +193,39 @@ public IList<IProvider> Children
     [Serializable]
     public class ProviderCategoryConfiguration
     {
-        public string Name { get; set; }
+        public ProviderCategoryConfiguration()
+        {
+            Providers = new Dictionary<string, IProviderConfiguration>();
+        }
 
-        public IDictionary<string, IProviderConfiguration> Providers { get; set; }
+        internal ProviderCategoryConfiguration(string name) : this()
```

Comment:
Well, the class itself is public. In that case it could be used in any other assembly if it's provided through a public function -- even if not constructed. This may not happen, I don't know (should one care?).

But what would be the reason not to check parameters even overzealously and then remove checks if they hinder performance or clutter readability badly? To me it gives a feeling assurance the code is robust, especially whilst refactoring code and debugging and so uncovering previously untried execution paths which nevertheless should work. That is, the code don't run amok perhaps many layers down making me lose some time to find the actual reason why it went on amok. This particular reason may be simple, but then again, if you, me, or someone else reading somebody else's code could see checking is there, it's a clear signal what are the invariants right on the spot. If they aren't there, maybe it should warrant for more scrutiny and reading? Taking more time?